Server/Quality_of_Life_changer.Implementation/Handlers/CommandHandlers/AddUserCalendarCommandHandler.cs

```diff
@@ -0,0 +1,104 @@
+﻿namespace Quality_of_Life_changer.Implementation.Handlers.CommandHandlers;
+
+using Contracts.Commands;
+using Contracts.Exceptions;
+using Data;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+public class AddUserCalendarCommandHandler : BaseCommandHandler,
+    IRequestHandler<AddUserCalendarCommand>
+{
+    private readonly QolcDbContext _context;
+
+    public AddUserCalendarCommandHandler(QolcDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<Unit> Handle(AddUserCalendarCommand request,
+        CancellationToken cancellationToken)
+    {
+        if (!await IsFirstUserCalendarWithUrl(request.Url, request.OwnerId, cancellationToken))
+        {
+            throw new ValidationException("You already have a calendar with this url already exist");
```

Comment:
we discussed, that only name for user should be unique. no need to check for URL